### PR TITLE
Emergency Contact + Cleanup

### PR DESCRIPTION
**Pull request by mattwilson02**
This **Pull Request** involves a new page, as well as some code cleanup and organisations.

- **Emergency Contact** page: This is a form capturing the necessary data and storing under _NIP 78_ App Specific rules
- Moved optional **Lobstr Logo** to Layout component
- Created an enum for different _App Specific Tags_ sent with NIP 78 Nostr events
- Added a constant for the relays to connect to
- Extracted the ndk connector and the Signer creation to a React Hook

### CODE REVIEW

File: app/scuba-onboarding/emergency-contact/page.tsx [hunk 1]
```diff
@@ -1,10 +1,74 @@
...
+            name,
+            phoneNumber,
+            email,
+          },
+        }),
+        pubkey: publickey,
+        tags: [['d', AppSpecificTags.EmergencyContact]],
+      })
+
+      await event.publish()
```
Comment (RicardoBrito1938) [inline app/scuba-onboarding/emergency-contact/page.tsx:38]: can we have a try catch here? or does the api return any error?

File: components/Layout/index.tsx [hunk 2]
```diff
@@ -3,24 +3,36 @@ import { PropsWithChildren, useCallback } from 'react'
...
 
   return (
-    <main className="flex min-h-screen h-full flex-col bg-white ">
+    <main className="flex min-h-screen h-full flex-col bg-white">
       {canGoBack && (
         <div className="p-2" onClick={goBack}>
           <IoIosArrowBack className="text-black" />
         </div>
       )}
+      {logoSize && (
```
Comment (RicardoBrito1938) [inline components/Layout/index.tsx:31]: worth to use pattern composition, then we leave the size default 200

File: app/scuba-onboarding/emergency-contact/page.tsx [hunk 3]
```diff
@@ -1,10 +1,74 @@
...
+
+  const router = useRouter()
+  const { publickey } = useAppSelector(({ user }) => user)
+  const { ndk } = useNostr()
+
+  const onSubmit = useCallback(
+    async ({ name, phoneNumber, email }: FieldValues) => {
+      const event = new NDKEvent(ndk, {
+        kind: NDKKind.AppSpecificData,
+        created_at: Math.floor(new Date().getTime() / 1000),
```
Comment (RicardoBrito1938) [inline app/scuba-onboarding/emergency-contact/page.tsx:26]: we can use Date.now() here

also we can consider using dayjs here

File: app/scuba-onboarding/emergency-contact/page.tsx [hunk 4]
```diff
@@ -1,10 +1,74 @@
...
+
   return (
-    <Layout>
-      <h1>Emergency Contact</h1>
+    <Layout logoSize={200}>
+      <div className="item-center flex flex-col gap-4 align-middle w-full px-10">
+        <h1 className="text-2xl  font-semibold text-center font-heading text-primary-500 mb-2">
+          Emergency contact
+        </h1>
+        <input
```
Comment (RicardoBrito1938) [inline app/scuba-onboarding/emergency-contact/page.tsx:51]: similar input, time to make our input component, or make our design system clone

File: hooks/useNostr.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    () => new NDKPrivateKeySigner(privatekey),
+    [privatekey],
+  )
+
+  ndk.signer = privatekey ? signer : nip07signer
+
+  return {
+    ndk,
+  }
+}
```
Comment (RicardoBrito1938) [inline hooks/useNostr.ts:26]: this hook is sweet

**Review by RicardoBrito1938 — APPROVED** [review]

File: app/scuba-onboarding/emergency-contact/page.tsx (hunk 4 above)
Comment (mattwilson02) [reply app/scuba-onboarding/emergency-contact/page.tsx:51]: agreed, i was about to, but then i remembered design system, lets do it asap

**Pull request merged**